### PR TITLE
81 unit tests for internationalization

### PR DESCRIPTION
**Pull request by VSirvio**
**Changes:**
- Refactor the language selection UI element to its own React component in `frontend/components/LangSelect.jsx`
- Write unit tests for the `LangSelect` UI component

### CODE REVIEW

File: frontend/tests/components/LangSelect.test.jsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/* eslint-disable no-unused-vars */
+import { render, screen } from '@testing-library/react'
+import LangSelect from '../../src/components/LangSelect'
+import { expect, vi } from 'vitest'
```
Comment (amandahamynen) [inline frontend/tests/components/LangSelect.test.jsx:4]:
```suggestion
import { expect } from 'vitest'
```

**Review by amandahamynen — CHANGES_REQUESTED** [review]
Good job! The component ```LangSelect.jsx``` and the corresponding test file ```LangSelect.test.jsx``` look good to me and work as intended. However, after inspecting the component, I was wondering if it might be worth adding a test to verify that the cookie is set correcly, since the component handles this at line 5:

```js
document.cookie = `lang=${lang}; expires=${new Date(Date.now().valueOf() + 2592000000).toUTCString()}; path=/`
```

What are your thoughts on that?

I also noticed an unused import ```vi``` at line 4 in ```LangSelect.test.jsx```, which can be removed.

Lastly just as a heads up, before i can approve this PR, the conflicts in ```frontend/src/App.jsx``` need to be resolved.

File: frontend/tests/components/LangSelect.test.jsx (hunk 1 above)
Comment (VSirvio) [reply frontend/tests/components/LangSelect.test.jsx]: Well noticed!

**Comment by VSirvio** [comment]
> I also noticed an unused import `vi` at line 4 in `LangSelect.test.jsx`, which can be removed.
> 
> Lastly just as a heads up, before i can approve this PR, the conflicts in `frontend/src/App.jsx` need to be resolved.

Both fixed now.

**Pull request merged**